services/processing/flake_processing.py

```diff
@@ -20,77 +20,77 @@ def process_flake_for_repo_commit(
     repo_id: int,
     commit_id: str,
 ):
-    uploads = ReportSession.objects.filter(
-        report__report_type=CommitReport.ReportType.TEST_RESULTS.value,
-        report__commit__repository__repoid=repo_id,
-        report__commit__commitid=commit_id,
-        state__in=["processed", "v2_finished"],
-    )
-
-    curr_flakes = fetch_curr_flakes(repo_id)
-    new_flakes: dict[str, Flake] = dict()
-
-    rollups_to_update: list[DailyTestRollup] = []
-
-    flaky_tests = list(curr_flakes.keys())
+    with django_transaction.atomic():
```

Comment:
instead of indenting all of this code, how about moving the transaction to the caller (along with the trailing log and return statement)